presto-main/src/main/java/com/facebook/presto/type/DecimalSaturatedFloorCasts.java

```diff
@@ -95,4 +105,119 @@ private static BigInteger bigintToBigintFloorSaturatedCast(BigInteger value, int
         }
         return unscaledValue;
     }
+
+    public static final SqlScalarFunction DOUBLE_TO_DECIMAL_SATURATED_FLOOR_CAST = SqlScalarFunction.builder(DecimalSaturatedFloorCasts.class)
```

Comment:
you could maybe use https://github.com/prestodb/presto/pull/5583, if it is in release branch